code/game/objects/items/candle.dm

```diff
@@ -82,13 +92,14 @@
 		return
 	if(!infinite)
 		wax--
+		if(wax_index != SHORT_CANDLE) // It's not at its shortest
+			get_wax_index()
```

Comment:
```suggestion
			if(update_wax_index())
				update_icon(UPDATE_ICON_STATE)
```
If you make `update_wax_index` return if it was updated or not, then you can update the icon here. This way the `update_wax_index` method stays a bit purer as to what it does. It should only update the `wax_index`, not also update the icon.